torch/_inductor/codegen/cpp_prefix.h

```diff
@@ -643,7 +644,37 @@ void atomic_add_vec(T *addr, at::vec::VectorizedN<int64_t, NI> index, at::vec::V
 }
 #endif
 
-void mm_get_thread_blocking(
+std::tuple<std::shared_ptr<int64_t[]>, int> _get_factors(int64_t number) {
+  int count = 0;
+  for (int64_t i = std::sqrt(number); i > 0; --i) {
+    if (number % i == 0) {
+      count += 2;
+    }
+  }
+  auto factors = std::shared_ptr<int64_t[]>(new int64_t[count]);
+  int index = 0;
+  for (int64_t i = std::sqrt(number); i > 0; --i) {
+    if (number % i == 0) {
+      factors[index++] = number / i;
+      factors[index++] = i;
+    }
+  }
+  return std::make_tuple(factors, count);
+}
+
+std::tuple<std::shared_ptr<int64_t[]>, int> get_factors(int64_t number) {
+  thread_local std::map<int64_t, std::tuple<std::shared_ptr<int64_t[]>, int>> cache;
```

Comment:
nit: why do we need to set this cache as `thread_local`? If multiple threads are running workloads, I think these threads can share these factors since it only relates to the number of threads for division.

Reply:
> nit: why do we need to set this cache as `thread_local`? If multiple threads are running workloads, I think these threads can share these factors since it only relates to the number of threads for division. Is it due to the concern about initializing this cache with multi threads?

Yes, `thread_local` avoids the lock contention when multiple threads need to access and modify the cache.